integration/class-enable-mastodon-apps.php

```diff
@@ -233,4 +250,104 @@ public static function api_search( $search_data, $request ) {
 
 		return $search_data;
 	}
+
+	public function api_get_posts_query_args( $args ) {
+		if ( isset( $args['author'] ) && is_string( $args['author'] ) ) {
+			$uri = Webfinger_Util::resolve( $args['author'] );
+			if ( $uri && ! is_wp_error( $uri ) ) {
+				$args['activitypub'] = $uri;
+				unset( $args['author'] );
+			}
+		}
+
+		return $args;
+	}
+
+	public static function api_statuses_external( $statuses, $args ) {
+		if ( ! isset( $args['activitypub'] ) ) {
+			return $statuses;
+		}
+
+		$data = get_remote_metadata_by_actor( $args['activitypub'] );
+
+		if ( ! $data || is_wp_error( $data ) || ! isset( $data['outbox'] ) ) {
+			return $statuses;
+		}
+
+		$response = Http::get( $data['outbox'], true );
+		if ( is_wp_error( $response ) || wp_remote_retrieve_response_code( $response ) !== 200 ) {
+			return $statuses;
+		}
+
+		$outbox = json_decode( wp_remote_retrieve_body( $response ), true );
+		if ( ! $outbox || is_wp_error( $outbox ) || ! isset( $outbox['first'] ) ) {
+			return $statuses;
+		}
+
+		$account = self::get_account_for_actor( $args['activitypub'] );
+		if ( ! $account ) {
+			return $statuses;
+		}
+
+		$response = Http::get( $outbox['first'], true );
+		if ( is_wp_error( $response ) || wp_remote_retrieve_response_code( $response ) !== 200 ) {
+			return $statuses;
+		}
+		$posts = json_decode( wp_remote_retrieve_body( $response ), true );
+
+		$activitypub_statuses = array_map(
+			function ( $item ) use ( $account ) {
+				$object = $item['object'];
+				if ( ! isset( $object['type'] ) || 'Note' !== $object['type'] ) {
+					return null;
+				}
+
+				$status = new Status();
+				$status->id         = Mastodon_API::remap_url( $object['id'] );
+				$status->created_at = new DateTime( $object['published'] );
+				$status->content    = $object['content'];
+				$status->account    = $account;
+
+				if ( ! empty( $object['inReplyTo'] ) ) {
+					$status->in_reply_to_id = $object['inReplyTo'];
+				}
+
+				if ( ! empty( $object['visibility'] ) ) {
+					$status->visibility = $object['visibility'];
+				}
+
+				$status->uri = $object['url'];
+
+				if ( ! empty( $object['attachment'] ) ) {
+					$status->media_attachments = array_map(
+						function ( $attachment ) {
+							$media_attachment = new Media_Attachment();
+							$media_attachment->id = Mastodon_API::remap_url( $attachment['url'], $attachment );
+							$media_attachment->type = strtok( $attachment['mediaType'], '/' );
+
+							$media_attachment->url = $attachment['url'];
+							$media_attachment->preview_url = $attachment['url'];
+							$media_attachment->description = $attachment['name'];
+							$media_attachment->blurhash = $attachment['blurhash'];
```

Comment:
should we check if the keys exist? WordPress for example does not provide blurhashes yet?!?

Reply:
We could also use a default blurhash `LaS5;bKg%z,IofafjtkC.lw1RjNt` (see https://blurha.sh/):
![Screenshot 2024-04-12 at 12 11 01](https://github.com/Automattic/wordpress-activitypub/assets/203408/0e38c41d-9b31-43cb-829f-df060ebecad2)